Content.Server/_CD/VoteLink/VoteLinkSystem.cs

```diff
@@ -0,0 +1,210 @@
+                var options = msg.VoteData.Options
+                    .OrderBy(o => o.Option)
+                    .Select(o => o.Text)
+                    .ToList();
+
+                // Get sender's identity for announcement
+                var author = Loc.GetString("comms-console-announcement-unknown-sender"); // recycling
+                var tryGetIdentityShortInfoEvent =
+                    new TryGetIdentityShortInfoEvent(Transform(sender).MapUid ?? sender, sender);
+                RaiseLocalEvent(tryGetIdentityShortInfoEvent);
+                if (!string.IsNullOrEmpty(tryGetIdentityShortInfoEvent.Title))
+                    author = tryGetIdentityShortInfoEvent.Title;
+
+                // Create and dispatch announcement
+                var announcement = Loc.GetString(comp.Message,
+                    ("author", author),
+                    ("question", msg.VoteData.Question));
+
+                _chat.DispatchGlobalAnnouncement(
+                    announcement,
+                    Loc.GetString(comp.Title),
+                    announcementSound: comp.AnnouncementSound);
+
+                // Log it here so we can get the sender
+                _adminLogger.Add(LogType.Action,
+                    LogImpact.Medium,
+                    $"{ToPrettyString(sender)} Started a new vote: '{msg.VoteData.Question}' with options: {string.Join(", ", options)}");
+
+                // Try to start the vote
+                TryStartVote(
+                    station,
+                    msg.VoteData.Question,
+                    options,
+                    msg.VoteData.Duration);
+                break;
+        }
+    }
+
+    private void OnDatabaseInit(Entity<StationVoteDatabaseComponent> ent, ref ComponentInit args)
+    {
+        // Initialize if needed
+        UpdateVotes(ent);
+    }
+
+    public override void Update(float frameTime)
+    {
+        base.Update(frameTime);
+
+        var query = EntityQueryEnumerator<StationVoteDatabaseComponent>();
+        var curTime = _timing.CurTime;
+
+        while (query.MoveNext(out var uid, out var database))
+        {
+            if (database.ActiveVote == null || !database.ActiveVote.HasEnded(curTime))
+                continue;
+
+            // Move to history
+            database.VoteHistory.Add(database.ActiveVote);
+            if (database.VoteHistory.Count > database.MaxHistorySize)
+                database.VoteHistory.RemoveAt(0);
+
+            database.ActiveVote = null;
+
+            // Set the cooldown
+            database.Cooldown = curTime + database.CooldownDuration;
+
+            // Notify that vote has ended
+            var ev = new VoteEndedEvent(uid);
+            RaiseLocalEvent(ev);
+        }
+    }
+
+    /// <summary>
+    ///     Starts a new vote on the station
+    /// </summary>
+    public bool TryStartVote(EntityUid station, string question, List<string> options, TimeSpan? duration = null)
```

Comment:
Would taking a `VoteData` here be nicer? You are basically unpacking a `VoteData` when you call this only to create one again a few lines below.